stdlib/string.mli

```diff
@@ -260,7 +260,39 @@ val escaped : string -> string
     [escaped s] fails).
 
     @raise Invalid_argument if the result is longer than
-    {!Sys.max_string_length} bytes. *)
+    {!Sys.max_string_length} bytes.
+
+    @deprecated Use {!String.escape_ascii} instead. *)
+
+val escape_ascii : string -> string
+(** [escape_ascii s] is [s] with all characters outside the printable US-ASCII
+    range represented by escape sequences.
+
+    All characters outside the US-ASCII printable range \[0x20;0x7E\] are
+    escaped, as well as backslash (0x2F) and double-quote (0x22).
+
+    The function {!Scanf.unescaped} is a left inverse of [escaped],
+    i.e. [Scanf.unescaped (escaped s) = s] for any string [s] (unless
+    [escaped s] fails).
+
+    @raise Invalid_argument if the result is longer than
+    {!Sys.max_string_length} bytes.
+
+    @since 4.14.0 *)
+
+val escape : string -> string
+(** [escape s] is [s] with special characters replaced with escape sequences,
+    following the lexical conventions of OCaml.
+
+    The following characters are escaped:
+    double quote (['\"'], newline (['\n']), carriage return (['\r']),
+    tab (['\t\]), backspace (['\b']), DEL (0x7F),
+    and everything in the ['\x00' .. '\x1F'] range.
+
+    @raise Invalid_argument if the result is longer than
+    {!Sys.max_string_length} bytes.
+
+    @since 4.14.0 *)
```

Comment:
I prefer `escaped` to `escape`, but opinions from native English speakers are most welcome.

Also, I'd rather have two new functions `escaped_ascii` and `escaped_utf8`, with `escaped` defaulting to `escaped_ascii` for the time being, then eventually deprecated, then eventually defined as `escaped_utf8`.

Reply:
The current naming tends to favor adjectives/past participles, but not consistently. For example, `String.trim`, `String.capitalize`, `String.uncapitalize` are verbs.

This likely needs a broader and more general discussion. I don't have a strong opinion, I'll go with maintainers decision.

Reply:
I'm curious to hear from native English speakers.  I still like the idea of verbs being used for imperative functions (e.g. `print`, `output`) and nouns/adjectives for pure functions (e.g. `length`), but as you point out, we have many exceptions to this idea already.

Reply:
Related point: naming a function `escape_utf8` (verb) would be a bad idea because it sounds like it escapes unicode characters, while `escaped_utf8`makes it more clear that `utf8` is a modifier, not the object of the `escape` verb (or so I think).

Reply:
I agree with @xavierleroy in liking the pure/functional nature of `escaped` (although as he frequently improves and corrects my written English in documentation, I don't put much weight on it being my first language!)

To me, `Bytes.escape : bytes -> unit` mutates its argument, where `Bytes.escaped : bytes -> bytes` leaves the argument unmodified.

In terms of the inconsistency, having `String.escaped` and `String.uppercase` is one thing, having `String.escape` and `Scanf.unescaped` is another.

Reply:
I always thought of `uppercase` as an adjective here, not a verb (like `escaped`).